lib/chef/provider/package/habitat.rb

```diff
@@ -0,0 +1,196 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require_relative "../../http/simple"
+require_relative "../../json_compat"
+require_relative "../../exceptions"
+require_relative "../package"
+require_relative "../../resource/habitat/habitat_package"
+# Bring in needed shared methods
+
+class Chef
+  class Provider
+    class Package
+      class Habitat < Chef::Provider::Package
+        use "../../resource/habitat/habitat_shared"
+        provides :habitat_package
+
+        #
+        # TODO list for `hab pkg`:
+        #
+        # kinda sorta analogous to:
+        #   apt-cache search
+        #   dpkg -l
+        #   dpkg -r / dpkg -P (without depsolving?)
+        #   apt-get remove/purge (with depsolving?)
+        #
+        # - hab pkg search ruby
+        # - hab pkg info lamont-granquist/ruby
+        # - hab pkg info lamont-granquist/ruby/2.3.1
+        # - hab pkg info lamont-granquist/ruby/2.3.1/20160101010101
+        #   ^^^^^ these will all need client-side caches for the "universe" of the depot
+        # - hab pkg uninstall lamont-granquist/ruby
+        # - hab pkg uninstall lamont-granquist/ruby/2.3.1
+        # - hab pkg uninstall lamont-granquist/ruby/2.3.1/20160101010101
+        # - hab pkg list (localinfo?) lamont-granquist/ruby
+        # - hab pkg list (localinfo?) lamont-granquist/ruby/2.3.1
+        # - hab pkg list (localinfo?) lamont-granquist/ruby/2.3.1/20160101010101
+        #   ^^^^^ need a better name
+        #
+        # Probably also want to support installation of local packages
+        # Service resource supports running services from locally installed packages
+        # But we provide no way to handle installation
+
+        def load_current_resource
+          @current_resource = Chef::Resource::HabitatPackage.new(new_resource.name)
+          current_resource.package_name(strip_version(new_resource.package_name))
+
+          @candidate_version = candidate_versions
+          current_resource.version(current_versions)
+
+          current_resource
+        end
+
+        def install_package(names, versions)
+          names.zip(versions).map do |n, v|
+            opts = ["pkg", "install", "--channel", new_resource.channel, "--url", new_resource.bldr_url]
+            opts += ["--auth", new_resource.auth_token] if new_resource.auth_token
+            opts += ["#{strip_version(n)}/#{v}", new_resource.options]
+            opts += ["--binlink"] if new_resource.binlink
+            opts += ["--force"] if new_resource.binlink.eql? :force
+            hab(opts)
+          end
+        end
+
+        alias_method :upgrade_package, :install_package
+
+        def remove_package(names, versions)
+          # raise 'It is too dangerous to :remove packages with the habitat_package resource right now. This functionality should be deferred to the hab cli.'
+          names.zip(versions).map do |n, v|
```

Comment:
Where's the .zip method coming from here?

Reply:
This was one of the things @lamont-granquist was going to check out

Reply:
It looks to be a bug from what I can tell. it's always worked until a recent update to master(i wanna say 3.0) it part of ruby so it should just work.. [rubydoc](https://ruby-doc.org/core-3.0.1/Array.html#method-i-zip)

Reply:
it looks like at some point I had removed `use_multipackage_api` which caused the issue I was seeing with this. I have corrected that and things are back to working